### PR TITLE
[rdy] Refactor cured died

### PR DESCRIPTION
**Pull request by Alberth289346**
*Reorganizes and cleans up the cured/death code.

### CODE REVIEW

**Comment by Alberth289346** [comment]
Besides the above small Travis problem, I am currently looking at
https://github.com/CorsixTH/CorsixTH/blob/3c2f7e51e58a14529c67901623fefe80e566df81/CorsixTH/Lua/hospital.lua#L1608   versus https://github.com/CorsixTH/CorsixTH/blob/3c2f7e51e58a14529c67901623fefe80e566df81/CorsixTH/Lua/hospital.lua#L1879

There are differences, but no idea of the right direction.

There is also 
https://github.com/CorsixTH/CorsixTH/blob/3c2f7e51e58a14529c67901623fefe80e566df81/CorsixTH/Lua/hospital.lua#L1902

File: CorsixTH/Lua/hospital.lua [hunk 1]
```diff
@@ -257,27 +257,38 @@ function Hospital:Hospital(world, avail_rooms, name)
...
 function Hospital:msgKilled()
-  local msg_chance = math.random(1, 10)
-  if self.num_deaths > 1 then
-    if msg_chance < 4 and self.msg_counter > 10 then
+  self.world.ui:playSound("boo.wav") -- this sound is always heard
+
+  if self.num_deaths < 1 then -- First death is always reported.
+    self.world.ui.adviser:say(_A.information.first_death)
+
+  else if self.num_deaths > 1 and self.msg_counter > 10 then
```
Comment (lewri) [inline CorsixTH/Lua/hospital.lua:286]: Needs to be ``elseif`` here, probably cause of the travis error.

Comment (Alberth289346) [reply CorsixTH/Lua/hospital.lua:286]: Thanks!

**Comment by Alberth289346** [comment]
Limiting changes to the message and audio feedback, consistency between cured and died is left for another day.

**Pull request merged**